src/static/js/webmentions.js

```diff
@@ -0,0 +1,288 @@
+// Code related to parsing and showing webmentions
+
+// Gets the webmentions json data from current URL
+async function getWebmentions(targetURL) {
+  const apiURL = `https://webmention.io/api/mentions.jf2?perPage=500&target=${targetURL}`;
+  let mentions = []
+  try {
+    const response = await window.fetch(apiURL);
+    if (response.status >= 200 && response.status < 300) {
+      json = await response.json();
+      mentions = json.children;
+    } else {
+      console.error("Could not parse response", response.statusText);
+    }
+  } catch(error) {
+    console.error("Request failed", error);
+  }
+  return mentions;
+}
+
+// Format published date into human readable form
+function formatDate(dateString){
+  const options = { year: "numeric", month: "long", day: "numeric" }
+  return new Date(dateString).toLocaleDateString(undefined, options)
+}
+
+// Parse webmentions for individual category
+function parseMentions(webmentions, mentionType) {
+  let filteredMentions = []
+  webmentions.forEach(function(mention){
+    if (mention["wm-property"] == mentionType) {
+      filteredMentions.push(mention)
+    }
+  });
+  return filteredMentions;
+}
+
+// Renders webmention likes
+function renderLikes(likes) {
+  // Add the likes count to the likes-tab
+  document.querySelector('#likes-count').innerHTML = likes.length;
+  let likeHtmlElements = [];
+  likes.forEach(function(like) {
+    const likeHtml = `
+      <li class="webmention-likes-item">
+        <a
+          class="webmention-author"
+          href="${like["url"]}"
+          title="${like["author"]["name"]}"
+          aria-label="${like["author"]["name"]} liked this chapter"
```

Comment:
I wonder if it's worth decoupling the descriptive text ("... liked this chapter") from the script. For example, we could translate it by defining the text in the localized Jinja templates. To get the translation into the script, we could use `querySelector` to grab the invisible element out of the DOM.

@tunetheweb WDYT?

Reply:
Yeah we normally have a `data-` attribute for this to allow translated text to be picked up from the template.

For example:
https://github.com/HTTPArchive/almanac.httparchive.org/blob/ea67e4b8a40c1c712394f7412c26f866f9f982a2/src/static/js/almanac.js#L158-L159